rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,211 @@
+
+The use of TODS will slightly modify the overall architecture diagram from above (assuming that RTR reads data from TODS as well):
+
+```mermaid
+  flowchart BT
+    hastus["HASTUS Schedule"]
+	tods["TODS"]
+	glides["Glides"]
+	rtr["RTR"]
+	ocs["OCS via Trike"]
+	predictions["TripUpdates"]
+	vehicles["VehiclePositions"]
+	hastus -- Trips --> tods
+	tods -- Trips --> glides
+	tods -- Trips --> rtr
+	glides -. Trip Assignments .-> rtr
+	glides -. Trainsheet Edits .-> rtr
+	ocs -. Vehicle Locations .-> rtr
+	rtr -. Vehicle Locations .-> glides
+	rtr -. Publishes .-> predictions
+	rtr -. Publishes .-> vehicles
+```
+
+## Realtime data: Glides trainsheets and RTR
+
+In the first stage, RTR and Glides will continue to maintain their own separate trip assignment systems, but Glides will begin sending HASTUS-based trip IDs in trainsheet update events. For scheduled trips, the [trip key published in `glides.trips_updated` events](https://mbta.github.io/schemas/events/glides/com.mbta.ctd.glides.trips_updated.v1#tripkey) will need to be updated in a new version of the schema to include this ID. The `serviceDate` will still need to be present in order to map from HASTUS schedule IDs to GTFS service IDs. The [Glides code that generates the trip key for the event](https://github.com/mbta/glides/blob/fbe9578c8400b955304c74d34e672460a0d17a23/lib/glides_web/channels/trainsheet_channel.ex#L1324) already has access to the `ScheduledTrip` with its trip ID, so once we are importing the trip ID from HASTUS we will simply be able to reference that. This will be a backwards-incompatible change, requiring a new version of the schema.
+
+RTR will use the trip IDs from the trainsheet edit events when determining what trip ID to publish for a vehicle matching that trip. The trip matching logic itself will still live with RTR at this point, but if RTR believes that a train is a particular trip from the trainsheet, it should use the corresponding trip ID in the public GTFS-rt feed. However, if a train that is assigned to a scheduled trip from Glides changes state in such a way that it should be assigned to a different trip by RTR's logic (for instance: unexpectedly going off its current pattern, or changing directions), RTR is free to assign that train to a different trip of its choosing. RTR will still be free to create its own added trips as needed.
+
+Additionally, RTR will need to handle cases where current service as reflected in GTFS static does not match the HASTUS-based trainsheets in Glides, due to planned disruptions modeled by `gtfs_creator`. In these cases, the data will manifest as trainsheet update messages with HASTUS-based trip IDs that do not match up with the trip IDs currently running per the GTFS static schedule. This will continue to be a potential issue, even with trip matching handled in Glides, so it is explored in more depth in the [Reference-level explanation](#reference-level-explanation) section.
```

Comment:
That is my understanding as well, though I tend to worry about logic that tries to parse out details from IDs rather than treating them as atomic values.

Reply:
Agree that we shouldn't rely on it, but it's better than nothing.